### ISSUE TITLE
Feature Request: Allow option to active automatically when plugged in the power outlet

### ISSUE THREAD

**Issue by brodock** [newmarcel/KeepingYouAwake#104]
It would be nice to have it activate automatically the first time when you plug the laptop in the power outlet.

That should happen only right after it is plugged in, so if you want to disable it after you can just click the notification icon again.

**Comment by newmarcel** [comment]
That's a nice idea. I wonder if more people could find this useful?

I can imagine implementing this as optional behaviour that you can enable in the preferences.

**Comment by Swiftnesses** [comment]
I need this :)

**Comment by ThomDietrich** [comment]
Hello,
I need this as well. My device is plugged in most of the time and KeepingYouAwake is active all the time. However, it rightfully deactivates itself when the battery runs low. I want the application to re-activate as soon as plugged in (or charged to a certain level).

How about a solution on the Advanced settings tab: Reactivate when battery capacity is above:"?

![image](https://user-images.githubusercontent.com/2870104/77561099-8dcbb280-6ebe-11ea-9258-b425292e2b4f.png)

**Comment by teunis90** [comment]
Yes, I was surprised that there is an option for "Deactive when battery capacity is low" but "auto-disable/enable" is not implemented. I'm not an OSX developer myself, is there a system event that can be used to implement this behavior?

**Comment by sturza** [comment]
I will work on this.

**Comment by sturza** [comment]
The implementation activates the keep awake function in ±10 seconds after plugged in.

**Comment by zmeir** [comment]
Would love to see this implemented :)  
(Also, maybe a complementary feature to allow automatically deactivating when disconnected from power?)

**Comment by jmillerdesign** [comment]
Here's how I'd suggest adding this feature:
Right below "Deactivate when Low Power mode is enabled"
add another checkbox for "Activate when Low Power mode is disabled"
